src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.HttpLogging
+{
+    /// <summary>
+    /// Middleware that logs HTTP requests and HTTP responses.
+    /// </summary>
+    internal sealed class HttpLoggingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+        private readonly IOptionsMonitor<HttpLoggingOptions> _options;
+        private const int DefaultRequestFieldsMinusHeaders = 7;
+        private const int DefaultResponseFieldsMinusHeaders = 2;
+        private const string Redacted = "X";
+
+        /// <summary>
+        /// Initializes <see cref="HttpLoggingMiddleware" />.
+        /// </summary>
+        /// <param name="next"></param>
+        /// <param name="options"></param>
+        /// <param name="logger"></param>
+        public HttpLoggingMiddleware(RequestDelegate next, IOptionsMonitor<HttpLoggingOptions> options, ILogger<HttpLoggingMiddleware> logger)
+        {
+            _next = next ?? throw new ArgumentNullException(nameof(next));
+
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            _options = options;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Invokes the <see cref="HttpLoggingMiddleware" />.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>HttpResponseLog.cs
+        public async Task Invoke(HttpContext context)
+        {
+            var options = _options.CurrentValue;
+            RequestBufferingStream? requestBufferingStream = null;
+            Stream? originalBody = null;
+
+            if ((HttpLoggingFields.Request & options.LoggingFields) != HttpLoggingFields.None)
```

Comment:
I notice we don't check if Logging is enabled before we do any of this stuff. Seems like we should no?